### DISCUSSION TITLE
PostGIS db function works locally but not in cloud, GEOS error

### DISCUSSION THREAD

**Discussion by acao** [supabase/supabase#14013] [Questions]
so, this function works great locally with `npx supabase`, but not in the cloud.

```sql
drop function if exists "public"."findzones"(aoi geometry);
set check_function_bodies = off;
-- expose this query
CREATE OR REPLACE FUNCTION public.findzones(
    aoi geometry(POLYGON, 4326)
)
-- the classifying data, and some relevant quantitative data
RETURNS TABLE (
    -- this ID is incremental, specified by the qGIS export I used in reference-schema.sql migration
    zone_section_id integer,
    -- these are non-unique, human readable string identifiers used for classifying the AOI's carbon
    -- sequestration potential
    climate_zone varchar,
    fldd_zone varchar,
    geom geometry,
    intersection geometry
)
LANGUAGE plpgsql
as $$
BEGIN
RETURN query
    SELECT ogc_fid, clmt_zn, fldd_zn, geometry,
    -- retrieve the intersection geometry, so we can compute area and total % of the AOI later
    ST_intersection("public"."ipcczones"."geometry", aoi) as intersection
    FROM "public"."ipcczones"
    -- return only the geometries that match this vector map
    WHERE ST_intersects("public"."ipcczones"."geometry", aoi);
END;$$
```

the error I get on supabase:
```json
{
	"error": {
		"code": "XX000",
		"details": null,
		"hint": null,
		"message": "lwgeom_intersection_prec: GEOS Error: TopologyException: Input geom 0 is invalid: Ring Self-intersection at or near point -62 5.5 at -62 5.5"
	}
}
```

the db diff shows no differences, the schemas are completely synced up locally and remotely. others are able to get it working only locally as well.

is there perhaps a difference in the configuration for local postgres with docker and hosted postgres extensions?

(fwiw, the geometry columns in the schema referenced above both use `geometry(POLYGON, 4236)`

**Comment by GaryAustin1** [comment]
Are they same postgres version?
You could also check the extension versions with `select * FROM pg_extension where extname = 'postgis';`

**Reply by acao** [reply]
thank you! the versions are both 3.3.2 it seems

the only `pg_extension` columns for this query that differ between local and our hosted supabase are `extowner` and `extconfig` if that matters!

**Reply by acao** [reply]
any news on this? i would like to make sure future supabase users needing PostGIS support don't get stuck on this as well!

**Reply by GaryAustin1** [reply]
I doubt anyone is looking at this, unless it was reported as an issue.  Still seems strange as in general postgis does work in the cloud.  So either something specific about a feature you are using, or your data is different between local and cloud for some reason.
